internal/server/change_status.go

```diff
@@ -0,0 +1,78 @@
+package server
+
+import (
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/ministryofjustice/opg-go-common/template"
+	"github.com/ministryofjustice/opg-sirius-lpa-frontend/internal/sirius"
+)
+
+type ChangeStatusClient interface {
+	Case(sirius.Context, int) (sirius.Case, error)
+	EditCase(sirius.Context, int, sirius.CaseType, sirius.Case) error
+	GetAvailableStatuses(sirius.Context, int, sirius.CaseType) ([]string, error)
+}
+
+type changeStatusData struct {
+	XSRFToken string
+	Entity    string
+	Success   bool
+	Error     sirius.ValidationError
+
+	AvailableStatuses []string
+	NewStatus         string
+}
+
+func ChangeStatus(client ChangeStatusClient, tmpl template.Template) Handler {
+	return func(w http.ResponseWriter, r *http.Request) error {
+		caseID, err := strconv.Atoi(r.FormValue("id"))
+		if err != nil {
+			return err
+		}
+
+		caseType, err := sirius.ParseCaseType(r.FormValue("case"))
+		if err != nil {
+			return err
+		}
+
+		ctx := getContext(r)
+
+		caseitem, err := client.Case(ctx, caseID)
+		if err != nil {
+			return err
+		}
+
+		AvailableStatuses, err := client.GetAvailableStatuses(ctx, caseID, caseType)
```

Comment:
hate to be that guy, but can you lowercase this

Reply:
also in Go they tend to not use `Get` in names so would just have `client.AvailableStatuses`